go/lib.h

```diff
@@ -23,19 +23,23 @@ typedef enum RequestErrorType {
  */
 typedef struct ConnectionResponse {
   const void *conn_ptr;
-  const char *error_message;
-  RequestErrorType error_type;
+  const char *connection_error_message;
 } ConnectionResponse;
 
 /**
  * Success callback that is called when a Redis command succeeds.
+ *
+ * `channel_address` is the address of the Go channel used by the callback to send the error message back to the caller of the command.
+ * `message` is the value returned by the Redis command.
  */
 typedef void (*SuccessCallback)(uintptr_t channel_address, const char *message);
 
 /**
  * Failure callback that is called when a Redis command fails.
  *
- * `error` should be manually freed after this callback is invoked, otherwise a memory leak will occur.
+ * `channel_address` is the address of the Go channel used by the callback to send the error message back to the caller of the command.
+ * `error_message` is the error message returned by Redis for the failed command. It should be manually freed after this callback is invoked, otherwise a memory leak will occur.
```

Comment:
> It should be manually freed
by whom?  and how?
usually `free` means that the same allocator should free the data - this means that rust code should free rust allocations, and go code should free go allocations.